File: e-comerce-backend/springboot/src/main/java/com/cdweb/springboot/service/UserService.java
package com.cdweb.springboot.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.cdweb.springboot.entities.Product;
import com.cdweb.springboot.entities.User;
import com.cdweb.springboot.request.UserRequest;
import com.cdweb.springboot.response.UserResponse;

public interface UserService {

	// public User getUserById(Long userld);
	// public User getUserProfileByJwt(String jwt);
	// public User getUserByEmailAndPassword(String email, String password);
	   UserResponse createUser(UserRequest userRequest);
	   UserResponse getUserById(Long id);
    UserResponse updateUser(Long id, UserRequest userRequest);

    List<UserResponse> getAllUsers();

    void deleteUser(Long id);
}

File: e-comerce-fontend/src/services/userService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/users';

export const userService = {
  getAllUsers: async () => {
    const response = await axios.get(API_URL);
    return response.data;
  },
  getUserById: async (id) => {
    const response = await axios.get(`${API_URL}/${id}`);
    return response.data;
  },

  updateUser: async (id, userData) => {
    const response = await axios.put(`${API_URL}/${id}`, userData);
    return response.data;
  },

  deleteUser: async (id) => {
    const response = await axios.delete(`${API_URL}/${id}`);
    return response.data;
  }
};